src/overlays/actors/ovl_En_Wonder_Talk/z_en_wonder_talk.c

```diff
@@ -26,15 +28,217 @@ const ActorInit En_Wonder_Talk_InitVars = {
+                this->actor.textId = 0x5020;
+                this->unk_156 = 6;
+                this->fHeight = 0.0f;
+                osSyncPrintf("\x1b[35m☆☆☆☆☆ 注目座標\t       \t☆☆☆☆☆ %f\n\x1b[m", 0.0f);
+                this->unk_15C = 120.0f;
+                if ((gSaveContext.eventChkInf[1] & 0x2000) != 0) {
+                    Actor_Kill(&this->actor);
+                }
+                break;
+            case 5:
+                this->actor.textId = 0x501F;
+                this->unk_156 = 5;
+                this->fHeight = 0.0f;
+                this->unk_15C = 110.0f;
+                osSyncPrintf("\x1b[35m☆☆☆☆☆ 注目座標\t       \t☆☆☆☆☆ %f\n\x1b[m", 0.0f);
+                break;
+            default:
+                this->actor.textId = 0x7072;
+                this->unk_156 = 5;
+                break;
+        }
+
+        osSyncPrintf("\x1b[36m☆☆☆☆☆ this->actor.talk_message    ☆☆☆☆☆ %x\n\x1b[m", this->actor.textId);
+        this->actionFunc = func_80B3943C;
+    }
+}
+
+void func_80B3943C(EnWonderTalk* this, GlobalContext* globalCtx) {
+    s16 yawDiff;
+    s16 phi_v1;
+
+    this->unk_15A++;
+    if (this->unk_150 == 4 && (gSaveContext.eventChkInf[1] & 0x2000)) {
+        Actor_Kill(&this->actor);
+    } else if ((this->unk_154 < 0) || (Flags_GetSwitch(globalCtx, this->unk_154) == 0)) {
+        if ((func_8002F194(&this->actor, globalCtx))) {
+            if (this->unk_156 != 6) {
+                this->actionFunc = func_80B395F0;
+            } else {
+                if (this->unk_154 >= 0) {
+                    this->actor.flags &= -2;
+                    Flags_SetSwitch(globalCtx, this->unk_154);
+                }
+                this->actionFunc = func_80B391CC;
+            }
+        } else if (!(this->unk_15C < this->actor.xzDistFromLink)) {
+            yawDiff = (this->actor.yawTowardsLink - this->actor.posRot.rot.y);
+            if (yawDiff >= 0) {
+                phi_v1 = yawDiff;
+            } else {
+                phi_v1 = -yawDiff;
+            }
+            if (phi_v1 < 0x4000) {
+                if (this->unk_15A >= 2) {
+                    osSyncPrintf("\n\n");
+                    osSyncPrintf("\x1b[33m☆☆☆☆☆ セーブ情報\t\t☆☆☆☆☆ %d\n\x1b[m", this->unk_154);
+                    osSyncPrintf("\x1b[35m☆☆☆☆☆ 種類インデックス\t☆☆☆☆☆ %d\n\x1b[m", this->unk_150);
+                    osSyncPrintf("\x1b[36m☆☆☆☆☆ 実質メッセージ種類     %x\n\x1b[m", this->actor.textId);
+                    osSyncPrintf("\x1b[32m☆☆☆☆☆ 指定範囲               %d\n\x1b[m", this->actor.posRot.rot.z);
+                    osSyncPrintf("\n\n");
+                }
+                this->unk_15A = 0;
+                func_8002F2CC(&this->actor, globalCtx, this->unk_15C);
+            }
+        }
+    }
+}
+
+void func_80B395F0(EnWonderTalk* this, GlobalContext* globalCtx) {
+    if (this->unk_156 == func_8010BDBC(&globalCtx->msgCtx)) {
+        if (func_80106BC8(globalCtx)) {
+            if (this->unk_154 >= 0) {
+                this->actor.flags &= -2;
```

Comment:
Pretty sure this works
```suggestion
                this->actor.flags &= ~1;
```

Reply:
I agree, this should work and is the preferred way to write it.